client/datasync-client/src/DataManager.ts

```diff
@@ -179,8 +179,7 @@ export class DataManager {
     const options = _.defaults(userOptions, defaultOptions);
     function forceSyncThenStop(pendingUpdateQueueSize) {
       if (pendingUpdateQueueSize === 0) {
-        self.stop().then(Bluebird.resolve);
-        return;
+        return self.stop().then(Bluebird.resolve);
       }
```

Comment:
I know its not your change but worth improving: then is not needed

Reply:
btw the cleanest way to wrap a promise in a bluebird one is `return Bluebird.resolve(promisefn())`